hub/tasks/x_event_source.py

```diff
@@ -0,0 +1,164 @@
+import os
+from contextlib import asynccontextmanager
+from functools import partial
+
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from apscheduler.triggers.interval import IntervalTrigger
+from nearai.shared.client_config import DEFAULT_MODEL
+
+from hub.api.v1.auth import AuthToken
+from hub.api.v1.thread_routes import RunCreateParamsBase, ThreadModel, _create_thread, create_run
+from hub.tasks.twitter_client import get_latest_mentions
+
+RESET_BLOCK_ID_ON_START = True  # reusing setting name for now
+
+# Path to store the last read tweet id
+TWEET_STATUS_FILE = "last_tweet.txt"
+
+scheduler = AsyncIOScheduler()
+
+# todo pull from registry 'integration' item
+EVENT_SOURCE_CONFIG = {"user_id": "1867270324649160704"}
+
+# todo pull from agent metadata
+LISTENER_CONFIG = {"agent_name": "flatirons.near/nearsecretagent/0.0.1"}
+
+
+# Function to get the last read tweet timestamp from a file
+async def get_last_tweet_timestamp():
```

Comment:
Shall we do it user specific? I assume we will have agents will the different frequency of mentions read.

Reply:
I think we want to use this to get all tweets since the last time we fetched tweets. I haven't implemented that though since I was testing that the agent handled duplicate tweets.